src/Dali/RedSharp.Dali.ViewModel/ImageItem.cs

```diff
@@ -0,0 +1,195 @@
+        {
+            if (rawBytes == null || rawBytes.Length == 0)
+                throw new ArgumentNullException(nameof(rawBytes), "Buffer is null or empty.");
+
+            _cache = rawBytes;
+        }
+
+        // For future use. Maybe this approach is faster...
+        /*public ImageItem(Memory<byte> memory)
+        {
+            MemoryHandle mem = memory.Pin();
+
+            unsafe
+            {
+                Image.Load(new ReadOnlySpan<byte>(mem.Pointer, memory.Length));
+            }
+        }*/
+        #endregion
+
+        #region Public Properties
+
+        /// <summary>
+        /// Stores fully loaded image. Should be set only for image on work area.
+        /// </summary>
+        public Image Image { get; private set; }
+
+        /// <summary>
+        /// Path to image file. Might be path to non existing file (for further saving).
+        /// </summary>
+        public string Path
+        {
+            get
+            {
+                return _path;
+            }
+            set
+            {
+                this.RaiseAndSetIfChanged(ref _path, value);
+            }
+        }
+
+        /// <summary>
+        /// Reduced image to show in list box.
+        /// </summary>
+        public Image Preview
+        {
+            get
+            {
+                if (_preview == null)
+                {
+                    using (Image fullImage = Image.Load(Cache))
+                    {
+                        _preview = fullImage.Clone(image => image.Resize(new ResizeOptions() { Mode = ResizeMode.Max, Size = new SixLabors.Primitives.Size(256, 256) }));
+                    }
+                }
+
+                return _preview;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets is item is currently selected in the list.
+        /// </summary>
+        public bool IsSelected
+        {
+            get
+            {
+                return _isSelectd;
+            }
+            set
+            {
+                this.RaiseAndSetIfChanged(ref _isSelectd, value);
+            }
+        }
+        #endregion
+
+        #region Private Properties
+        /// <summary>
+        /// Incupsulates logic for buffer reading. Returns cached images or tries to load it from saved path.
+        /// </summary>
+        private byte[] Cache
+        {
+            get
+            {
+                if (CacheImages || _cache != null)
+                {
+                    return _cache;
+                }
+                else
+                {
+                    if (string.IsNullOrEmpty(_path) || !File.Exists(_path))
+                        throw new FileNotFoundException("File path is invalid. Cannot load image.");
+
+                    using (FileStream file = File.OpenRead(_path))
+                    {
+                        byte[] buffer = new byte[file.Length];
+                        file.Read(buffer, 0, (int)file.Length);
+                        return buffer;
```

Comment:
You have similar code in the constructor, maybe be better to have one single method to read bytes?

Reply:
Ok. I will add one.

Reply:
Thanks a lot, and, by the way, my comments is a non block for merge - it's just a recommendation, so you can merge your changes without my approval in any time as you want